### ISSUE TITLE
New lint for: `panic!("{:?}");`

### ISSUE THREAD

**Issue by mcarton** [Manishearth/rust-clippy#518]
Surprisingly `rustc` does not error or even warn on:
```
% cat a.rs
fn main() {
    panic!("{:?}");
}
% rustc a.rs 
% echo $!
0
```

This is because when `panic!` has only one argument, it does not go through `format_args!`.
I guess one could write a lint that checks for the first argument to a call to `std::rt::begin_unwind`. I could do that if you think it's worth it. Or maybe you think that should be reported on the std repo and fixed there?

Oh, by the way, [clippy itself is affected by that](https://github.com/Manishearth/rust-clippy/blob/a65a7770b3e16fbe5e350c670d852c2696ce2907/src/consts.rs#L78) :sweat_smile:.

**Comment by birkenfeld** [comment]
Since this will not fail, but produce unexpected results, it's probably a good candidate for clippy.

But let's see if @Manishearth thinks it should go into rustc.

**Comment by llogiq** [comment]
I don't think this should stop us from developing it within clippy – we can move it into rustc later if we deem it appropriate.

**Comment by Manishearth** [comment]
So the rustc policy is mostly "no new lints". They may upstream lints from clippy or whatever if they get used a lot; or if a lint is really good and can only be done within the compiler. Still, it should be rare that a lint is added.

And we should put lints in clippy either way, and _later_ make the case for upstreaming to rust.
